tests/ptf/fabric.ptf/test.py

```diff
@@ -272,12 +295,75 @@ def runTest(self):
         testutils.verify_packets(self, exp_pkt_2to1, [self.port1])
 
 
+class ArpBroadcastTest(FabricTest):
+    def runArpBroadcastTest(self, tagged_ports, untagged_ports):
+        mac_addr_mask = ":".join(["ff"] * 6)
+        vlan_id = 10
+        next_id = vlan_id
+        mcast_group_id = vlan_id
+        all_ports = set(tagged_ports + untagged_ports)
+        arp_pkt = testutils.simple_arp_packet(pktlen=76)
+        # Account for VLAN header size in total pktlen
+        vlan_arp_pkt = testutils.simple_arp_packet(vlan_vid=vlan_id, pktlen=80)
+
+        for port in tagged_ports:
+            self.set_ingress_port_vlan(port, True, vlan_id, vlan_id)
+        for port in untagged_ports:
+            self.set_ingress_port_vlan(port, False, 0, vlan_id)
+        self.add_bridging_entry(vlan_id, arp_pkt.dst, mac_addr_mask, next_id)
+        self.add_next_multicast(next_id, mcast_group_id)
+        self.add_mcast_group(mcast_group_id, all_ports)
+        for port in untagged_ports:
+            self.set_egress_vlan_pop(port, vlan_id)
+
+        for inport in all_ports:
+            pkt_to_send = vlan_arp_pkt if inport in tagged_ports else arp_pkt
+            testutils.send_packet(self, inport, str(pkt_to_send))
+            # Packet should be received on all ports expect the ingress one.
+            verify_tagged_ports = set(tagged_ports)
+            verify_tagged_ports.discard(inport)
+            for tport in verify_tagged_ports:
+                testutils.verify_packet(self, vlan_arp_pkt, tport)
+            verify_untagged_ports = set(untagged_ports)
+            verify_untagged_ports.discard(inport)
+            for uport in verify_untagged_ports:
+                testutils.verify_packet(self, arp_pkt, uport)
+            testutils.verify_no_other_packets(self)
```

Comment:
you should call this outside of the loop

Reply:
Done